File: app/src/main/java/org/mifos/mobilebanking/ui/widgets/ChargeWidgetService.java
package org.mifos.mobilebanking.ui.widgets;

import android.content.Intent;
import android.widget.RemoteViewsService;

/**
 * ChargeWidgetService is the {@link RemoteViewsService} that will return our RemoteViewsFactory
 */
public class ChargeWidgetService extends RemoteViewsService {
    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return new ChargeWidgetDataProvider(this, intent);
    }
}

File: app/src/main/java/org/mifos/mobile/ui/widgets/ChargeWidgetService.kt
package org.mifos.mobile.ui.widgets

import android.content.Intent
import android.widget.RemoteViewsService

/**
 * ChargeWidgetService is the [RemoteViewsService] that will return our RemoteViewsFactory
 */
class ChargeWidgetService : RemoteViewsService() {

    override fun onGetViewFactory(intent: Intent): RemoteViewsFactory {
        return ChargeWidgetDataProvider(this)
    }
}